common/common.go

```diff
@@ -83,8 +83,9 @@ var (
 	errHTTPClientInvalid       = errors.New("custom http client cannot be nil")
 )
 
-// NilGuard returns an ErrNilPointer with the type of the first nil argument
-func NilGuard(ptrs ...any) (errs error) {
+// NilGuard returns an ErrNilPointer with the type of the first nil value found in the arguments.
+// It returns immediately upon encountering the first nil, which may not be the first argument.
```

Comment:
🚧 Change request:
* Existing comment was both incorrect for the original implementation, and valid for the new implementation
* This explantation kinda feels overcooked - It just returns for the first nil argument. End of story.
```suggestion 
// NilGuard returns an ErrNilPointer with the type of the first nil argument
```